risc0/zkvm/src/host/api/convert.rs

```diff
@@ -481,19 +557,87 @@ impl TryFrom<pb::core::InnerReceipt> for InnerReceipt {
             pb::core::inner_receipt::Kind::Composite(inner) => Self::Composite(inner.try_into()?),
             pb::core::inner_receipt::Kind::Groth16(inner) => Self::Compact(inner.try_into()?),
             pb::core::inner_receipt::Kind::Succinct(inner) => Self::Succinct(inner.try_into()?),
-            pb::core::inner_receipt::Kind::Fake(inner) => Self::Fake {
-                claim: inner.claim.ok_or(malformed_err())?.try_into()?,
-            },
+            pb::core::inner_receipt::Kind::Fake(inner) => Self::Fake(inner.try_into()?),
+        })
+    }
+}
+
+// NOTE: InnerReceipt and InnerAssumptionReceipt are the same type in protobuf.
+// In Rust, they are distinct types becaue Rust needs to size everything on the
+// stack and e.g. SuccinctReceipt<ReceiptClaim> and SuccinctReceipt<Unknown>
+// have different sizes. Protobuf handles this without issue.
```

Comment:
I wonder if this will bite us in the end when we want to start making changes to one or the other. What's nice about having separate types in protobuf is that they can vary independently in a backwards-compatible way.

Reply:
Part of my assumption here is that you can "fork" the types later as needed without introducing a backwards incompatibility at the protobuf layer. I _think_ this is true, but I'm not 100% sure.

I'd be happy to split this out now

Reply:
Ah, if we can fork later in a compatible way, then this seems fine to me.